### PR TITLE
Add docker registry mirror

### LINKED ISSUE #570: Add support for docker registry-mirrors
Some environments have dedicated docker mirrors to speed up the image pull process.

Docker supports image mirrors via a Flag: `--registry-mirror` or `daemon.json` option:
```json
{
  "registry-mirrors": ["https://registry.docker-cn.com"]
}
```

This will make docker pull all images using that registry, instead of the one which gets specified via the Kubernetes manifest. Which makes this option perfect for transparent image caching!

To pass through that option to the docker daemon, the machine-controller needs a new flag, like [--node-insecure-registries](https://github.com/kubermatic/machine-controller/blob/master/cmd/controller/main.go#L204), which then gets passed through to the nodes.

### PR DESCRIPTION
**Pull request by themue**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #570 

**Special notes for your reviewer**:

```release-note
- Adds flag -node-registry-mirrors to set mirrors for Docker image registries
```

### CODE REVIEW

File: pkg/controller/machine/machine.go [hunk 1]
```diff
@@ -107,17 +107,19 @@ type Controller struct {
...
-	// Translates to --cluster-dns on the kubelet
+	// Translates to --cluster-dns on the kubelet.
 	ClusterDNSIPs []net.IP
 	// If set, this proxy will be configured on all nodes.
 	HTTPProxy string
-	// If set this will be set as NO_PROXY on the node
+	// If set this will be set as NO_PROXY on the node.
 	NoProxy string
-	// If set, this image registry will be used for pulling all required images on the node
+	// If set, this image registry will be used for pulling all required images on the node.
```
Comment (mrIncompetent) [inline pkg/controller/machine/machine.go]: Could you fix this comment? I screwed that up in the initial PR :(

`If set, those registries will be configured as insecure on the container runtime.`

Comment (themue) [reply pkg/controller/machine/machine.go]: Done. Will rush in with next push.

File: pkg/userdata/centos/provider.go [hunk 2]
```diff
@@ -268,11 +268,16 @@ write_files:
...
 {{ containerRuntimeHealthCheckSystemdUnit | indent 4 }}
 
-{{- if .InsecureRegistries }}
+{{- if or .InsecureRegistries .RegistryMirrors }}
 - path: /run/containers/registries.conf
   permissions: "0644"
   content: |
-    INSECURE_REGISTRY="--insecure-registry {{ .InsecureRegistries | join " --insecure-registry "}}"
+    {{- if .InsecureRegistries}}
+    INSECURE_REGISTRY="{{range .InsecureRegistries}}--insecure-registry {{.}} {{end}}"
```
Comment (alvaroaleman) [inline pkg/userdata/centos/provider.go]: Why did you change this expression? The prior one did not leave a trailing space.

File: cmd/controller/main.go [hunk 3]
```diff
@@ -331,8 +338,16 @@ func main() {
 	}
 
 	for _, registry := range strings.Split(nodeInsecureRegistries, ",") {
-		if strings.TrimSpace(registry) != "" {
-			runOptions.node.InsecureRegistries = append(runOptions.node.InsecureRegistries)
+		trimmedRegistry := strings.TrimSpace(registry)
```
Comment (alvaroaleman) [inline cmd/controller/main.go]: Nit: Mind using here and below the shorted ` if trimmedRegistry := strings.TrimSpace(registry); trimmgedRegeistry != "" {`

File: cmd/controller/main.go [hunk 4]
```diff
@@ -180,7 +181,8 @@ type controllerRunOptions struct {
 	node machinecontroller.NodeSettings
 }
 
-func main() {
+// parseFlags encapsulates the parsing of all flags.
+func parseFlags() {
```
Comment (alvaroaleman) [inline cmd/controller/main.go]: Why?

File: pkg/userdata/centos/provider.go (hunk 2 above)
Comment (themue) [reply pkg/userdata/centos/provider.go]: Henrik asked me to change it to a loop.

File: cmd/controller/main.go (hunk 3 above)
Comment (themue) [reply cmd/controller/main.go]: Will change.

File: cmd/controller/main.go (hunk 4 above)
Comment (themue) [reply cmd/controller/main.go]: After adding only my one flag the cyclomatic complexity test failed. 🤦‍♂ Moving it out into this function made it pass again. 

IMHO the better solution would be a refactoring into a `Main` type with helping methods and a major `Run()` method called in `main()`. This would also be possible to be unit tested (needing a lot of stubbing). But that would be a different issue.

Comment (alvaroaleman) [reply cmd/controller/main.go]: Or just remove the cyclomatic complecity linter? It doesn't care about nesting at all it just counts number of branches within a function which IMHO is a pretty useless metric.

Comment (themue) [reply cmd/controller/main.go]: Would have no problem with it. Got some similar messages in my private code too where IMHO the splitting with the needed data structures and/or arguments would lead to an even more unmaintainable code.

But we need a general agreement then.

Comment (alvaroaleman) [reply cmd/controller/main.go]: Can you just add an exception for this specific case so we can get this pull merged?

Comment (themue) [reply cmd/controller/main.go]: Done.

File: pkg/userdata/centos/provider.go (hunk 2 above)
Comment (mrIncompetent) [reply pkg/userdata/centos/provider.go]: Yes, that `strings.Join` was rather counter intuitive & meant we always set `--insecure-registry`.
That worked well as long as we only had one variable - Since we now have 2 which can be set individually that would not have worked
